File: study/ts/script/data/CampRaceUserData.ts
import { BaseData } from './BaseData';
import { HeroConst } from '../const/HeroConst';
import { AvatarDataHelper } from '../utils/data/AvatarDataHelper';
import { HeroDataHelper } from '../utils/data/HeroDataHelper';
let schema = {};
schema['id'] = [
    'number',
    0
];

schema['base_id'] = [
    'number',
    0
];

schema['avatar_base_id'] = [
    'number',
    0
];

schema['name'] = [
    'string',
    ''
];

schema['officer_level'] = [
    'number',
    0
];

schema['score'] = [
    'number',
    0
];

schema['power'] = [
    'number',
    0
];

export interface CampRaceUserData {
    getId(): number
    setId(value: number): void
    getLastId(): number
    getBase_id(): number
    setBase_id(value: number): void
    getLastBase_id(): number
    getAvatar_base_id(): number
    setAvatar_base_id(value: number): void
    getLastAvatar_base_id(): number
    getName(): string
    setName(value: string): void
    getLastName(): string
    getOfficer_level(): number
    setOfficer_level(value: number): void
    getLastOfficer_level(): number
    getScore(): number
    setScore(value: number): void
    getLastScore(): number
    getPower(): number
    setPower(value: number): void
    getLastPower(): number
}
export class CampRaceUserData extends BaseData {
    public static schema = schema;

    public clear() {
    }
    public reset() {
    }
    public getCoverId() {
        let avartarBaseId = this.getAvatar_base_id();
        let coverId = this.getBase_id();
        if (avartarBaseId > 0) {
            coverId = AvatarDataHelper.getAvatarConfig(avartarBaseId).hero_id;
        }
        return coverId;
    }
    public getLimitLevel() {
        let limitLevel = 0;
        let heroBaseId = this.getBase_id();
        let info = HeroDataHelper.getHeroConfig(heroBaseId);
        if (info.type == 1) {
            let avatarBaseId = this.getAvatar_base_id();
            let limit = AvatarDataHelper.getAvatarConfig(avatarBaseId).limit;
            if (limit == 1) {
                return HeroConst.HERO_LIMIT_RED_MAX_LEVEL;
            }
        }
        return limitLevel;
    }

    getLimitRedLevel() {
        var limitRedLevel = 0;
        var heroBaseId = this.getBase_id();
        var info = HeroDataHelper.getHeroConfig(heroBaseId);
        if (info.type == 1) {
            var avatarBaseId = this.getAvatar_base_id();
        }
        return limitRedLevel;
    }
}

File: study/lua/src/app/data/CampRaceUserData.lua
local BaseData = require("app.data.BaseData")
local CampRaceUserData = class("CampRaceUserData", BaseData)

local schema = {}
schema["id"] = {"number", 0}
schema["base_id"] = {"number", 0}
schema["avatar_base_id"] = {"number", 0}
schema["name"] = {"string", ""}
schema["officer_level"] = {"number", 0}
schema["score"] = {"number", 0}
schema["power"] = {"number", 0}
CampRaceUserData.schema = schema

function CampRaceUserData:ctor(properties)
    CampRaceUserData.super.ctor(self, properties)
end

function CampRaceUserData:clear()

end

function CampRaceUserData:reset()
	
end

function CampRaceUserData:getCoverId()
	local avartarBaseId = self:getAvatar_base_id()
	local coverId = self:getBase_id()
	if avartarBaseId > 0 then
		coverId = require("app.utils.data.AvatarDataHelper").getAvatarConfig(avartarBaseId).hero_id
	end
	return coverId
end

function CampRaceUserData:getLimitLevel()
	local limitLevel = 0
	local heroBaseId = self:getBase_id()
	local info = require("app.utils.data.HeroDataHelper").getHeroConfig(heroBaseId)
	if info.type == 1 then --主角
		local avatarBaseId = self:getAvatar_base_id()
		local limit = require("app.utils.data.AvatarDataHelper").getAvatarConfig(avatarBaseId).limit
		if limit == 1 then
			return require("app.const.HeroConst").HERO_LIMIT_RED_MAX_LEVEL
		end
	end
	return limitLevel
end

function CampRaceUserData:getLimitRedLevel()
	local limitRedLevel = 0
	local heroBaseId = self:getBase_id()
	local info = require("app.utils.data.HeroDataHelper").getHeroConfig(heroBaseId)
	if info.type == 1 then --主角
		local avatarBaseId = self:getAvatar_base_id()
		-- 变身卡暂不支持红升金
		-- local limit = require("app.utils.data.AvatarDataHelper").getAvatarConfig(avatarBaseId).limit
		-- if limit == 1 then
		-- 	return require("app.const.HeroConst").HERO_LIMIT_RED_MAX_LEVEL
		-- end
	end
	return limitRedLevel
end

return CampRaceUserData